src/main/java/games/strategy/engine/lobby/client/login/LobbyLogin.java

```diff
@@ -158,7 +187,22 @@ public void notifyFailedLogin(final String message) {
               final Map<String, String> props = new HashMap<>();
               props.put(LobbyLoginValidator.REGISTER_NEW_USER_KEY, Boolean.TRUE.toString());
               props.put(LobbyLoginValidator.EMAIL_KEY, createAccount.getEmail());
+              // TODO: Don't send the md5-hashed password once the lobby is updated
               props.put(LobbyLoginValidator.HASHED_PASSWORD_KEY, MD5Crypt.crypt(createAccount.getPassword()));
+              if (challengProperties.get(LobbyLoginValidator.RSA_PUBLIC_KEY) != null) {
+                try {
+                  final PublicKey publicKey =
+                      KeyFactory.getInstance(LobbyLoginValidator.RSA).generatePublic(new X509EncodedKeySpec(
+                          Base64.getDecoder().decode(challengProperties.get(LobbyLoginValidator.RSA_PUBLIC_KEY))));
+                  final Cipher cipher = Cipher.getInstance(LobbyLoginValidator.RSA);
+                  cipher.init(Cipher.ENCRYPT_MODE, publicKey);
+                  props.put(LobbyLoginValidator.ENCRYPTED_PASSWORD_KEY, Base64.getEncoder().encodeToString(
+                      cipher.doFinal(Util.sha512(createAccount.getPassword()).getBytes(StandardCharsets.UTF_8))));
+                } catch (InvalidKeySpecException | NoSuchAlgorithmException | IllegalBlockSizeException
+                    | BadPaddingException | InvalidKeyException | NoSuchPaddingException e) {
```

Comment:
Same comment as above.  Actually, this whole block of code looks like a paste from above.  If so, can we extract it to a new method?